src/camera_utils.py

```diff
@@ -43,7 +58,23 @@ def compute_image_footprint_on_surface(camera: Camera, distance_from_surface: fl
     Returns:
         np.ndarray: [footprint_x, footprint_y] in meters.
     """
-    raise NotImplementedError()
+
+    # Convert distance to mm
+    distance_mm = distance_from_surface * 1000
+
+    #Small Triangle: Base: sensor_size_x_mm Height: focal length
+    #Big Triangle: Base: footprint_x_mm Height: distance_mm
+    # triangles are similar, the ratios of their corresponding sides are equal
+    # (sensor_size_x_mm) / focal length = (footprint_x_mm) / (distance_mm)
+    focal_length_x = compute_focal_length_in_mm(camera)[0]  #convert focal length into mm
+    focal_length_y = compute_focal_length_in_mm(camera)[1]
+
+    # Compute the image footprint in mm
+    footprint_x = (camera.sensor_size_x_mm * distance_mm) / focal_length_x
+    footprint_y = (camera.sensor_size_y_mm * distance_mm) / focal_length_y
+
+
+    return np.array([footprint_x/1000, footprint_y/1000]) # convert back to meters
```

Comment:
Nit: since you want to return the footprint in meters, there is no need to convert the input distance from meters to mm